src/frontend/apps/impress/src/features/docs/doc-tree/hooks/useActionableMode.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useRef } from 'react';
+
+import { SELECTORS } from '../dom-selectors';
+
+export type ActionableNodeLike = {
+  isFocused?: boolean;
+  focus?: () => void;
+};
+
+export const useActionableMode = (
+  node: ActionableNodeLike,
+  isMenuOpen?: boolean,
+) => {
+  const actionsRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    // Handles F2 to focus the first actionable element in the actions area, except when a modal is open
+    const toActions = (e: KeyboardEvent) => {
+      if (e.key !== 'F2' || document.querySelector(SELECTORS.MODAL)) {
+        return;
+      }
+
+      // Only react if the node is currently focused
+      if (!node?.isFocused) {
+        return;
+      }
+
```

Comment:
I am not fond of coupling with `dom-selectors`. If a modal is opened I suppose the node would not be focused anymore, so is it not enough that ?
```suggestion
      if (e.key !== 'F2' || !node?.isFocused) {
        return;
      }
```